### ISSUE TITLE
"as_user" has been deprecated by Slack but Botkit still uses it

### ISSUE THREAD

**Issue by GaryTowers** [howdyai/botkit#1943]
### What are you trying to achieve or the steps to reproduce?
I'm trying to send a message from my bot with a custom avatar using the `icon_url` property

````javascript
await bot.reply({
   text: "Hey this is me",
   icon_url: "http://lorempixel.com/48/48/" //<-- Botkit reads this and forces the use of a deprecated 'as_user' field
});
````

### What was the result you received?
I received an error message form the Slack API that it is tied with the `as_user` when using `icon_url` in the reply object.
```
code:"slack_webapi_platform_error"
data:Object {ok: false, error: "invalid_arguments", deprecated_argument: "as_user", …}
message:"An API error occurred: invalid_arguments"
stack:"Error: An API error occurred: invalid_arguments
```

### What did you expect?
I expected that the `icon_url` field could be used without the `as_user` field, because the `icon_url` is now dependent over Slack's [new granular bot permissions](https://api.slack.com/authentication/basics#public).

However the Botkit Slack Adapter is still assuming the field `as_user` is required to be sent as `false` when `icon_url` is used on replies [(slack_adapter.ts file)](https://github.com/howdyai/botkit/blob/7bcb10033c1c9cfa319e398b0f012aafcb207275/packages/botbuilder-adapter-slack/src/slack_adapter.ts#L343) as you would do with a previous generation **classic bot**.

Slack mentions that the `as_user` field [(may not be used on new generation bots)](https://api.slack.com/methods/chat.postMessage#arg_as_user)

My recommendation to fix this would be to remove the [(enforcement of the IF statement)](https://github.com/howdyai/botkit/blob/7bcb10033c1c9cfa319e398b0f012aafcb207275/packages/botbuilder-adapter-slack/src/slack_adapter.ts#L343) setting `as_user` to `false` when setting the field `icon_url` and leave it to the developer to manually include the `as_user` on their reply object if ever needed.

### Context:

- Botkit version: 4.8.1
- Messaging Platform: Slack
- Node version: 13.7.0
- Os: macOS Mojave

**Comment by benbrown** [comment]
This would be a breaking change for many bots that are NOT using granular scopes.

Any solution to this will have to consider the new `oauth_version` configuration option.

**Comment by GaryTowers** [comment]
Thanks for pointing this out @benbrown I updated the PR to consider using `oauth_version` instead and make the fix backwards compatible with v1

**Comment by benbrown** [comment]
Resolved in 4.9
